app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt

```diff
@@ -300,204 +86,281 @@ fun SeekerProfileScreen(
     onDispose { activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED }
   }
   // Collect the user profile from the StateFlow
+  val user by authViewModel.user.collectAsState()
   val userProfile by viewModel.seekerProfile.collectAsState()
-  val scrollState = rememberScrollState()
+  user?.let { viewModel.getUserProfile(it.uid) }
 
-  // Display the profile information if it's available
-  userProfile?.let { profile ->
-    Scaffold(
-        modifier = Modifier.background(colorScheme.background),
-        backgroundColor = colorScheme.background,
-        topBar = {
-          Column(modifier = Modifier.background(colorScheme.background)) {
-            Text(
-                text = "Profile",
-                style =
-                    TextStyle(
-                        fontSize = 24.sp,
-                        fontWeight = FontWeight.Bold,
-                        color = colorScheme.onBackground),
-                modifier =
-                    Modifier.padding(start = 16.dp, top = 16.dp)
-                        .verticalScroll(scrollState)
-                        .background(colorScheme.background)
-                        .testTag("ProfileTitle"), // testTag for the title
-            )
-            Spacer(modifier = Modifier.height(16.dp))
+  var fullName by remember { mutableStateOf("") }
+  var email by remember { mutableStateOf("") }
 
-            TopAppBar(
-                modifier = Modifier.testTag("ProfileTopBar"),
-                backgroundColor = colorScheme.primary, // Match background color from Figma
-                contentPadding = PaddingValues(16.dp)) {
-                  // Row layout for profile image, name, email, and edit icon
+  LaunchedEffect(userProfile) {
+    fullName = userProfile.name
+    email = userProfile.email
+  }
+
+  // State for logout dialog
+  var showLogoutDialog by remember { mutableStateOf(false) }
+
+  // Display the profile information if it's available
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = {
+              Text(
+                  "Profile",
+                  modifier = Modifier.testTag("ProfileTitle"),
+                  fontWeight = FontWeight.Bold,
+              )
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() },
```

Comment:
Would you happen to know why: when clicking on the goBack button after having reached the `SeekerProfileScreen` from the `SeekerMap`, we get stuck and the goBack button does not allow us to return to the other screens?

Reply:
This is due to a problem in our implementation of `navigateTo()` in `NavigationActions.kt`. This will be addressed through #316

Reply:
Perfect, thanks! Then it's approved!